tests/by-util/test_mv.rs

```diff
@@ -1443,6 +1444,29 @@ fn test_mv_dir_into_file_where_both_are_files() {
         .stderr_contains("mv: cannot stat 'a/': Not a directory");
 }
 
+#[test]
+fn test_mv_same_case_insensitive_file() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let file1 = "a.txt";
+    at.touch(file1);
+    #[cfg(target_os = "linux")]
```

Comment:
Are the BSDs case-insensitive as well? I think this might actually also be more complicated because this might also depend on the file system. I'm not entirely sure though.

Reply:
Yeah this is also where im a bit out of my league, because handling cross-platforms is so difficult, specially given that I cant test what is working or not. I am really wondering if we should even have different behaviors on different platforms, which has been my biggest issue :(. Thats why I have left it to @fdncred and other more smart people to figure out way forward

Reply:
I'd like to see at least the FreeBSD CI pass (it seems to be cancelled on the last run). Then we'll just see if anyone complains 😄

Reply:
I'd vote for consistency across all platforms as much as possible.